integration/service.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * CF7_SMTP context class.
+ *
+ * @package   cf7_smtp
+ * @author    Erik Golinelli <erik@codekraft.it>
+ * @copyright 2022 Erik
+ * @license   GPL 2.0+
+ * @link      https://modul-r.codekraft.it/
+ */
+
+
+if ( ! class_exists( 'WPCF7_Service' ) ) {
+	return;
+}
+
+use WPCF7_Service as GlobalWPCF7_Service;
+Class WPCF7_SMTP extends GlobalWPCF7_Service {
+
+    private static $instance;
+
+    public $options;
+
+    public static function get_instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+    public function __construct() {
+	    $this->options = get_option( 'cf7-smtp-options' );
+
+        $integration = 'cf7-smtp';
+        add_action('load-' . $integration, array($this, 'wpcf7_load_integration_page'), 10, 0);
+
+    }
+
+	public function get_title() {
+		return __( 'SMTP', 'Simple Mail Transfer Protocol' );
+	}
+
+    public function is_active() {
+        return $this->options['enabled'];
+	}
+
+	public function get_categories()
+	{
+		return array('email_services');
+	}
+
+    public function icon() {
+		echo '<div class="integration-icon">' . file_get_contents(CF7_SMTP_PLUGIN_ROOT . 'public/icon.svg') . '</div>';
+	}
+
+
+	public function link() {
+		return wpcf7_link(
+			'https://wordpress.org/plugins/cf7-smtp/',
+			'cf7-smtp'
+		);
+	}
+
+	public function admin_notice( $message = '' ) {
+	}
+
+    protected function menu_page_url( $args = '' ) {
+		$args = wp_parse_args( $args, array() );
+
+		$url = menu_page_url( 'wpcf7-integration', false );
+		$url = add_query_arg( array( 'service' => 'cf7-smtp' ), $url );
+
+		if ( ! empty( $args ) ) {
+			$url = add_query_arg( $args, $url );
+		}
+
+		return $url;
+	}
+
+
+    public function load( $action = '' ) {
+		if ( 'setup' == $action and 'POST' == $_SERVER['REQUEST_METHOD'] ) {
```

Comment:
| Detected usage of a possibly undefined superglobal
     |       | array index: $_SERVER['REQUEST_METHOD']. Use isset()
     |       | or empty() to check the index exists before using
     |       | it
     |       | (WordPress.Security.ValidatedSanitizedInput.InputNotValidated)